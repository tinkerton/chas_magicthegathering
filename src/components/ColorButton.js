import React from 'react';


const ColorButton =(props) => {
  
  const styles = ({
    manaButton: {
      display: 'flex',
      flexDirection: 'row',
      alignItems: 'center',
      boxSizing: 'border-box',
      borderRadius: '4px 4px 0px 0px',
      minWidth: '16%',
      textAlign: 'center',
      position: 'relative',
      borderBottom: '1px solid #CCC',
      background: '#F7F7F7',
      margin:'0px 4px 40px 0',
      height:'44px',
    },
    manaImage: {
        margin: '0px 0px 20px 0px',
        maxWidth:'20px',
        background: '#FFF',
        padding: '4px',
        borderRadius: '100px',
        position: 'absolute',
        top:'8px',
        left:'10px',
       
    },
    manaLabel: {
      fontFamily: 'Arial',
      fontWeight:'bold',
      textTransform: 'uppercase',
      fontSize: '12px',
      color:'#333',
      position: 'absolute',
      top:'15px',
      left:'45px',

    }
  })

    const {value,name,imagePath} = props.button;
    return (
        <div 
          style={styles.manaButton} 
          key={name} 
          button={value} 
          onClick={() => props.onChange({value})}>
        <img 
          style={styles.manaImage} 
          src={imagePath} 
          alt={name}
        />
        <span style={styles.manaLabel}>{name}</span>
      </div>
    );
  
 }
export default ColorButton;