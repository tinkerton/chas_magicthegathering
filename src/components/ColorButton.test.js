import React from "react";
import renderer from 'react-test-renderer';
import ColorButton from '../components/ColorButton';

it('renders correctly', () => {
    const buttonProps = {value: "B",  imagePath:"assets/images/B.svg" };
    const Button = renderer.create(<ColorButton key="Black" button={buttonProps}/>).toJSON();
    expect(Button).toMatchSnapshot();
});