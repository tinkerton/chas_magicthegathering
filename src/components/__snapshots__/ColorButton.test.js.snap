// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  button="B"
  onClick={[Function]}
  style={
    Object {
      "alignItems": "center",
      "background": "#F7F7F7",
      "borderBottom": "1px solid #CCC",
      "borderRadius": "4px 4px 0px 0px",
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "row",
      "height": "44px",
      "margin": "0px 4px 40px 0",
      "minWidth": "16%",
      "position": "relative",
      "textAlign": "center",
    }
  }
>
  <img
    src="assets/images/B.svg"
    style={
      Object {
        "background": "#FFF",
        "borderRadius": "100px",
        "left": "10px",
        "margin": "0px 0px 20px 0px",
        "maxWidth": "20px",
        "padding": "4px",
        "position": "absolute",
        "top": "8px",
      }
    }
  />
  <span
    style={
      Object {
        "color": "#333",
        "fontFamily": "Arial",
        "fontSize": "12px",
        "fontWeight": "bold",
        "left": "45px",
        "position": "absolute",
        "textTransform": "uppercase",
        "top": "15px",
      }
    }
  />
</div>
`;
